packages/java/parser-jvm-core/src/test/java/dev/hilla/parser/models/jackson/JacksonPropertyModelTests.java

```diff
@@ -0,0 +1,214 @@
+    public void should_HaveCorrectType(JacksonPropertyModel model,
+            String name) {
+        assertEquals(ctx.getType(name), model.getType());
+    }
+
+    @DisplayName("It should pass equality check")
+    @ParameterizedTest(name = ModelProvider.testNamePattern)
+    @ArgumentsSource(ModelProvider.class)
+    public void should_PassEqualityCheck(JacksonPropertyModel model,
+            String name) {
+        var expected = JacksonPropertyModel.of(ctx.getReflectionOrigin(name));
+
+        assertEquals(model, model);
+        assertEquals(expected, model);
+        assertNotEquals(model, new Object());
+    }
+
+    @DisplayName("It should provide common model class")
+    @ParameterizedTest(name = ModelProvider.testNamePattern)
+    @ArgumentsSource(ModelProvider.class)
+    public void should_ProvideCommonModelClass(JacksonPropertyModel model,
+            String name) {
+        assertEquals(model.getCommonModelClass(), JacksonPropertyModel.class);
+    }
+
+    @DisplayName("It should provide correct origin")
+    @ParameterizedTest(name = ModelProvider.testNamePattern)
+    @ArgumentsSource(ModelProvider.class)
+    public void should_ProvideCorrectOrigin(JacksonPropertyModel model,
+            String name) {
+        assertEquals(ctx.getReflectionOrigin(name), model.get());
+        assertTrue(model.isReflection());
+    }
+
+    @DisplayName("It should provide correct property info")
+    @ParameterizedTest(name = ModelProvider.testNamePattern)
+    @ArgumentsSource(ModelProvider.class)
+    public void should_ProvideCorrectPropertyInfo(JacksonPropertyModel model,
+            String name) {
+        assertEquals(ctx.getNameAfterRenaming(name), model.getName());
+        assertEquals(ClassInfoModel.of(Sample.class), model.getOwner());
+    }
+
+    @DisplayName("It should provide property annotations")
+    @ParameterizedTest(name = ModelProvider.testNamePattern)
+    @ArgumentsSource(ModelProvider.class)
+    public void should_ProvidePropertyAnnotations(JacksonPropertyModel model,
+            String name) {
+        assertEquals(ctx.getAnnotations(name), model.getAnnotations());
+    }
+
+    @DisplayName("It should provide property field (if there is one)")
+    @ParameterizedTest(name = ModelProvider.testNamePattern)
+    @ArgumentsSource(ModelProvider.class)
+    public void should_ProvidePropertyField(JacksonPropertyModel model,
+            String name) {
+        assertEquals(ctx.getField(name).isPresent(), model.hasField());
+        assertEquals(ctx.getField(name), model.getField());
+    }
+
+    @DisplayName("It should provide property getter (if there is one)")
+    @ParameterizedTest(name = ModelProvider.testNamePattern)
+    @ArgumentsSource(ModelProvider.class)
+    public void should_ProvidePropertyGetter(JacksonPropertyModel model,
+            String name) {
+        assertEquals(ctx.getGetter(name).isPresent(), model.hasGetter());
+        assertEquals(ctx.getGetter(name), model.getGetter());
```

Comment:
Well, I don't really want to test internals of Jackson, so it's just about covering the implementation of the model. Just in case anything is changed in the model code.

Reply:
Can't you just compare the getter name to a string?

Reply:
Yep, you're right. Let's do it this way.